example/app/config.yml

```diff
@@ -81,17 +81,52 @@ default:
     subnets:
       - '{{AWS_SUBNET}}'
 
+  alarms:
+    CPUHigh:
+      alarm_action: ScaleOutScalingPolicy
+      threshold: '50'
+    CPULow:
+      alarm_action: ScaleInScalingPolicy
+      comparison_operator: LessThanThreshold
+      threshold: '50'
+
   ecs:
     instanceType: t2.medium
     desiredInstances: 1
+    maxInstances: 2
+    minInstances: 1
     availabilityZone: us-east-1b
     amiid: ami-a7a242da
     publicIp: true
     restartTasksOnDeploy: true
     docker:
       username: cumulususer
+    autoscaling_policies:
+      ScaleOut:
+        policy_type: SimpleScaling
+        adjustment_type: ChangeInCapacity
+        scaling_adjustment: 1
+      ScaleIn:
+        policy_type: SimpleScaling
+        adjustment_type: ChangeInCapacity
+        scaling_adjustment: -1
+      Step:
+        policy_type: StepScaling
+        adjustment_type: ChangeInCapacity
+        step_adjustments:
+          - MetricIntervalLowerBound: 0
+            ScalingAdjustment: 1
+          - MetricIntervalUpperBound: 0
+            ScalingAdjustment: -1
+      Target:
+        policy_type: TargetTrackingScaling
+        adjustment_type: ChangeInCapacity
+        target_value: 50
+        predefined_metric_type:
+          PredefinedMetricType: ASGAverageCPUUtilization
```

Comment:
I think we discussed that CPU utilization is probably not the right metric in most cases, so I'm wary of having it be the example.  What if the tasks are fully utilized by I/O bound and not CPU bound (which I would expect to be the case most of the time)?  Scaling would never trigger, no matter how backed up the queue was.

Reply:
Changed to NetworkIn https://github.com/nasa/cumulus/pull/836/commits/bca1394d2ab511ffa7b3094b9b0178f6729e8313

Reply:
Is that right, though?  What if tasks are not network bound?

These two metrics feel like a hack around what we actually want to scale on, which is one of the following:

- percentage of tasks utilized vs idle.
- number of step function executions waiting for an available runner to pick up their task

Reply:
These metrics are helpful.
* Tasks utilized vs idle: Tasks have states of pending, running and stopped ([task lifecycle](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_life_cycle.html)). Tasks that are running as activities I believe are constantly running.
* Number of step function executions waiting: I think we can use `ActivitiesWaiting = ActivitiesScheduled - ActivitiesStarted` - if ActivitiesWaiting (a new metric) is > 0 we probably want to scale up the desired number of tasks for the associated service. https://docs.aws.amazon.com/step-functions/latest/dg/procedure-cw-metrics.html

Do we also want an example of scaling services and / or ec2 instances based on service memory utilization?